cdc/split.cc

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mutation.hh"
+#include "schema.hh"
+
+#include "split.hh"
+
+namespace cdc {
+
+bool should_split(const mutation& base_mutation, const schema& base_schema) {
```

Comment:
In the future I would like to refactor everything here and do the same thing I did in my previous version of the PR (https://github.com/scylladb/scylla/pull/5803), where I do not create a vector of mutations which is then `transform`ed in CDC, but instead use the `batch` struct directly in `transform`. This would mean we always call `extract_changes` and `should_split` will be gone.

We decided to abandon the previous PR because it did a lot of changes in `transform` which would conflict with changes made by other people. The plan is to merge this PR before the next release, without any major changes to `transform`, and then to refactor in a later PR, with the final result looking like in the previous abandoned PR.

So while I agree with the code duplication issue, I think we shouldn't worry about it now.